### PR TITLE
Fix inconsistency in soft lepton taginfos naming

### PR DESCRIPTION
**Pull request by ghost**
Fixing an inconsistency in the soft lepton taginfos naming, dating back to 2007:

https://github.com/cms-cvs-history/DataFormats-BTauReco/commit/1a967352b82098b95b16b2914ae8a90cd230680e#diff-e1aaed9bb312dbfd1dee141f852b4e78

Due to this inconsistency, the quantities:

sip2d, sip3d

are used as IP significances, while they are meant to be the signed IP values. These quantities are filled now with the IP values and two new taginfo variables are added for the significances themselves:

sip2dsig, sip3dsig

This fix solves an error-prone naming ambiguity.

This PR fixes also a bug in the use of brackets in SoftPFElectronTagInfoProducer.cc, where the original:

if(std::abs(properties.sip3d>MaxSip3D)) continue

was actually meant to be:

if(std::abs(properties.sip3d)>MaxSip3D) continue

This fix produces small differences with respect to the original version of the code, though not significantly affecting the performance.

### CODE REVIEW

**Comment by ferencek** [comment]
@imarches, can you also update `DQMOffline/RecoB/interface/SoftLeptonTagPlotter.h` and `DQMOffline/RecoB/src/SoftLeptonTagPlotter.cc`. Thanks.

File: RecoBTag/SoftLepton/plugins/SoftPFElectronTagInfoProducer.cc [hunk 1]
```diff
@@ -109,7 +113,7 @@ void SoftPFElectronTagInfoProducer::produce(edm::Event& iEvent, const edm::Event
   			properties.ratioRel = recoelectron->p4().Dot(jetRef->p4()) / pjet.Mag2();
   			properties.p0Par    = boostedPPar(recoelectron->momentum(), jetRef->momentum());
 			properties.elec_mva    = recoelectron->mva_e_pi();
-			 if(std::abs(properties.sip3d>MaxSip3D)) continue;
+			 if(std::abs(properties.sip3dsig>MaxSip3Dsig)) continue;
```
Comment (slava77) [inline RecoBTag/SoftLepton/plugins/SoftPFElectronTagInfoProducer.cc]: does this make sense?

Comment (ferencek) [inline RecoBTag/SoftLepton/plugins/SoftPFElectronTagInfoProducer.cc]: Good catch! I guess the original intention was to have `std::abs(properties.sip3dsig)>MaxSip3Dsig`. In practice I suspect the impact of this bug is negligible but still it should be fixed. Thanks again for catching it.

Comment (imarches) [inline RecoBTag/SoftLepton/plugins/SoftPFElectronTagInfoProducer.cc]: Yes, I'll add a commit to fix it. Yes, the impact should be negligible, as the cut is set by default at 200, that's why it passed unnoticed until now.

**Comment by imarches** [comment]
Hallo @ferencek yes I can also add the changes to the DQM to this same PR. I'll work at it this afternoon.

**Comment by slava77** [comment]
should we expect https://github.com/cms-sw/cmssw/pull/11247#issuecomment-140193218 to be addressed as well?

**Comment by imarches** [comment]
Hello @slava77 yes, please, wait a moment for the DQM things, I will address them right away here, as I'm about it anyway. Thanks! I will make the commit in a while.

**Comment by imarches** [comment]
Dear @slava77 now it should be it. @acaudron already had a look at the DQM part. Thanks.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by deguio** [comment]
+1

**Comment by cvuosalo** [comment]
@imarches:
Please add to the PR description that you fixed the "abs(a > b) to abs(a) > b" bug. This fix produces differences from the baseline, unlike the rest of the PR, so it is important to note.

**Comment by cvuosalo** [comment]
+1

For #11247 12d57feb1d931301edef9d2a7bdfe8f95686e2ea

Fixing a bug in SoftPFElectronTagInfoProducer.cc and renaming of  signed IP significance variables used for soft lepton tag infos.

The code changes are satisfactory, and Jenkins [tests](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11247/8092/summary.html) against baseline CMSSW_7_6_X_2015-09-13-1100 show no significant differences. The bug fix makes tiny changes in a few quantities related to the combined MVA and soft PF electron b taggers.

**Comment by imarches** [comment]
@cvuosalo I have added this explanation to the description. Thanks

**Comment by davidlange6** [comment]
+1

**Pull request merged**